File: Documentation/5.4/Samples/python/Indexes/Querying/ExplorationQueries.py
from examples_base import ExampleBase, Employee


class ExplorationQueries(ExampleBase):
    def setUp(self):
        super().setUp()

    def test_exploration_queries(self):
        with self.embedded_server.get_document_store("ExplorationQueries") as store:
            # filter in a collection query
            with store.open_session() as session:
                # region exploration-queries_1.3
                result = session.advanced.raw_query(
                    "from Employees as e " "filter e.Address.Country = 'USA' " "filter_limit 500", Employee
                ).single()
                # endregion

                # filter in an index query
                # region exploration-queries_2.3
                emp = (
                    session.advanced.raw_query(
                        "from Employees as e "
                        "where e.Title = $title "
                        "filter e.Address.Country = $country "
                        "filter_limit $limit",
                        Employee,
                    )
                    .add_parameter("title", "Sales Representative")
                    .add_parameter("country", "USA")
                    .add_parameter("limit", 500)
                    .single()
                )
                # endregion

                # filter and projection
                # region exploration-queries_3.3
                emp3 = session.advanced.raw_query(
                    "from Employees as e "
                    "filter startsWith(e.FirstName, 'A') "
                    "select { FullName: e.FirstName + ' ' + e.LastName }",
                    Employee,
                )
                # endregion

File: Documentation/5.4/Samples/php/Indexes/Querying/ExplorationQueries.php
<?php

namespace RavenDB\Samples\Indexes\Querying;

use RavenDB\Documents\DocumentStore;
use RavenDB\Samples\Infrastructure\Orders\Employee;

class ExplorationQueries
{
    public function explorationQueries(): void
    {
        $store = new DocumentStore();
        try {
            // filter in a collection query
            $session = $store->openSession();
            try {

                # region exploration-queries_1.1
                $result = $session->query(Employee::class)
                    ->filter(function($f) { return $f->equals("Address.Country", "USA"); }, 500)
                    ->singleOrDefault();
                # endregion

                # region exploration-queries_1.2
                $result = $session->advanced()->documentQuery(Employee::class)
                    ->filter(function($p) { return $p->equals("Address.Country", "USA"); }, limit: 500)
                    ->singleOrDefault();
                # endregion

                # region exploration-queries_1.3
                $result = $session->advanced()
                        ->rawQuery(
                            Employee::class,
                            "from Employees as e " .
                               "filter e.Address.Country = 'USA' " .
                               "filter_limit 500")
                ->singleOrDefault();
                # endregion
            } finally {
                $session->close();
            }

            // filter in an index query
            $session = $store->openSession();
            try {
            # region exploration-queries_2.1
            $emp = $session->query(Employee::class)
                ->whereEquals("Title", "Sales Representative")
                ->filter(function($f) { return $f->equals("Address.Country", "USA"); }, 500)
                ->singleOrDefault();
            # endregion

            # region exploration-queries_2.2
            $emp = $session->advanced()->documentQuery(Employee::class)
                  ->whereEquals("Title", "Sales Representative")
                  ->filter(function($p) { return $p->equals("Address.Country", "USA"); }, limit: 500)
                  ->singleOrDefault();
            # endregion

            # region exploration-queries_2.3
            $emp = $session->advanced()->rawQuery(Employee::class,
                "from Employees as e" .
                 "where e.Title = \$title" .
                 "filter e.Address.Country = \$country" .
                 "filter_limit \$limit")
                ->addParameter("title", "Sales Representative")
                ->addParameter("country", "USA")
                ->addParameter("limit", 500)
                ->singleOrDefault();
            # endregion
            } finally {
                $session->close();
            }

            // filter and projection
            $session = $store->openSession();
            try {
            # region exploration-queries_3.1
            $emp1 = $session
                ->query(Employee::class)
                ->filter(function($f) { return $f->equals("Address.Country", "USA"); }, 500)
                ->selectFields(null,  "Name", "Address.City", "Address.Country")
                ->toList();

                // Results:
                // ========

                // * Results include all companies with country = 'USA'
                //   Each resulting object contains only the selected fields.
                //
                // * No auto-index is created.
            # endregion

            # region exploration-queries_3.3
            $emp3 = $session->advanced()->rawQuery(
                Employee::class,
                "from Companies " .
                    "filter Address.Country == 'USA'" .
                    "select Name, Address.City, Address.Count`"
            );
            # endregion
            } finally {
                $session->close();
            }
        } finally {
            $store->close();
        }
    }
}
